llvm/lib/Transforms/Scalar/MemCpyOptimizer.cpp

```diff
@@ -265,7 +265,10 @@ void MemsetRanges::addRange(int64_t Start, int64_t Size, Value *Ptr,
     I->Start = Start;
     I->StartPtr = Ptr;
     I->Alignment = Alignment;
-    I->MaxAlignmentOffset = (I->MaxAlignmentOffset + Size) % I->MaxAlignment;
+    if (I->MaxAlignmentOffset != 0)
+      I->MaxAlignmentOffset = (I->MaxAlignmentOffset + Size) % I->MaxAlignment;
+    else
+      I->MaxAlignmentOffset = 0;
```

Comment:
Hmm, we are checking if the `MaxAlignmentOffset` is `0`, and if not we are initializing the variable, if yes we are again setting it to `0`.

Reply:
Yes you're right this is redundant! Thanks